python/ray/air/_internal/remote_storage.py

```diff
@@ -95,10 +95,25 @@ def get_fs_and_path(
         return fs, path
 
     try:
+        # in case of hdfs filesystem, if uri don't have the netloc part below will
+        # failed with hdfs access error.  For example 'hdfs:///user_folder/...' will
+        # fail, while only 'hdfs://namenode_server/user_foler/...' will work
+        if parsed.scheme == "hdfs" and parsed.netloc == "":
+            raise Exception(
```

Comment:
Since it will fail when calling from_uri() for HDFS missing netloc, do you still need to raise this exception? Can it just catch that exception raised from from_uri()?

Reply:
from_uri() can also take other type of address, such as local or s3 address.  So capturing exception over there will be more complicated.  We will need to sort out all possible scenarios.

I intend to make the change here to minimize the logic differences from the original code.